torch/distributed/distributed_c10d.py

```diff
@@ -2322,8 +2322,20 @@ def send(
         group (ProcessGroup, optional): The process group to work on. If None,
             the default process group will be used.
         tag (int, optional): Tag to match send with remote recv
+        group_dst (int, optional): Destination rank on ``group``. 
 
     """
+    if group_dst != None:
+        assert dst is None, "Can't specify both group_dst and dst"
+        # TODO if we want to encourage migration to 'group_dst' arg and deprecate dst arg, should we support using 'group_dst'
+        # even when 'group' is None (for default group)?
+        assert group is not None, "Must specify group if using group_dst"
+        dst = get_global_rank(group, group_dst)
```

Comment:
oops, this is dumb. i just realized that i can't implement group_dst as a function of global rank, since part of the reason for this PR is that 'group' might not be associated with the default-group at all, so get_global_rank function may not work.  We will need to invert the logic below so that 'group_dst' is the thing that we directly pass into the group api, and we convert dst into group_dst at the top if dst is passed in.  this will make the change a bit more invasive but still possible.

Reply:
There is a global to group map mapping: _world.pg_group_ranks

Reply:
yea, i realize that but the point is we want to not rely on any global stuff when using group_ argument.  This is to enable use cases where someone creates a ProcessGroup object directly and it's not registered with us.